File: 300.600/392.IsSubsequence.cpp
#include <iostream>
#include <string>
using namespace std;

class IsSubsequence{
public:
    bool isSubsequence(string s, string t) {
        int ss = s.size();
        int ps = 0;
        int st = t.size();
        int pt = 0;
        while (ps < ss && pt < st) {
            if (s[ps] == t[pt]) {
                ++ps;
            }
            ++pt;
        }
        return ps == ss;
    }
};

int main(int argc, char const *argv[])
{
    IsSubsequence cls;
    cout << boolalpha << cls.isSubsequence("abc", "ahbgdc") << " -> true" << endl;
    cout << boolalpha << cls.isSubsequence("axc", "ahbgdc") << " -> false" << endl;
    return 0;
}

File: go/300.500/392.IsSubsequence.go
package main

import "fmt"

func isSubsequence(s string, t string) bool {
	ss, ps, st, pt := len(s), 0, len(t), 0
	for ps < ss && pt < st {
		if s[ps] == t[pt] {
			ps++
		}
		pt++
	}
	return ps == ss
}

func main() {
	fmt.Println(isSubsequence("abc", "ahbgdc"), true)
	fmt.Println(isSubsequence("axc", "ahbgdc"), false)
}